File: 0007-reverse-integer/solution.java
class Solution {
    public int reverse(int num) {
        long res = 0;
        int temp = num;
        num = num < 0 ? -num : num;
        while (num > 0) {
            res = (res * 10) + (num % 10);
            // System.out.println(res); // uncomment this line if you want to see the steps
            num = Math.floorDiv(num, 10);
        }

        if(temp < 0) {
            return (res > (Math.pow(2, 31) - 1) || res < -(Math.pow(2, 31)) ? 0 : (int) -res);
        }
        return (res > (Math.pow(2, 31) - 1) || res < -(Math.pow(2, 31)) ? 0 : (int) res);
    }
}

File: 0007-reverse-integer/solution.go
import (
    "math"
)

func reverse(x int) int {
    result := 0
    for ; x != 0; {
        lastDigit := x % 10
        result = result * 10 + lastDigit

        if result > int(math.Pow(float64(2), float64(31)) - 1) {
            return 0
        } else if result < int(math.Pow(float64(-2), float64(31))) {
            return 0
        }

        x /= 10
    }

    return result
}
